docs/module_guides/evaluating/evaluating_evaluators_with_llamadatasets.md

```diff
@@ -0,0 +1,58 @@
+# Evaluating Evaluators with `LabelledEvaluatorDataset`'s
+
+The purpose of the llama-datasets is to provide builders the means to quickly benchmark
+LLM systems or tasks. In that spirit, the `LabelledEvaluatorDataset` exists to
+facilitate the evaluation of evaluators in a seamless and effortless manner.
+
+This dataset consists of examples that carries mainly the following attributes:
+`query`, `answer`, `ground_truth_answer`, and `reference_answer` along with some
+other supplementary attributes. The user flow for producing evaluations with this
+dataset consists of making predictions over the dataset with a provided LLM
+evaluator, and then computing metrics that measure goodness of evaluations by
+computationally comparing them to the corresponding references.
+
+Below is a snippet of code that makes use of the `EvaluatorBenchmarkerPack` to
+conveniently handle the above mentioned process flow.
+
+```python
+from llama_index.llama_dataset import download_llama_dataset
+from llama_index.llama_pack import download_llama_pack
+from llama_index.evaluation import CorrectnessEvaluator
+from llama_index.llms import Gemini
+from llama_index import ServiceContext
+
+# download dataset
+evaluator_dataset, _ = download_llama_dataset(
+    "MiniMtBenchSingleGradingDataset", "./mini_mt_bench_data"
+)
+
+# define evaluator
+gemini_pro_context = ServiceContext.from_defaults(
+    llm=Gemini(model="models/gemini-pro", temperature=0)
+)
+evaluator = CorrectnessEvaluator(service_context=gemini_pro_context)
+
+# download EvaluatorBenchmarkerPack and define the benchmarker
+EvaluatorBenchmarkerPack = download_llama_pack(
+    "EvaluatorBenchmarkerPack", "./pack"
+)
+evaluator_benchmarker = EvaluatorBenchmarkerPack(
+    evaluator=evaluators["gpt-3.5"],
+    eval_dataset=evaluator_dataset,
+    show_progress=True,
+)
+
+# produce the benchmark result
+benchmark_df = await evaluator_benchmarker.arun(
+    batch_size=5, sleep_time_in_seconds=0.5
+)
+```
+
+## The related `LabelledPairwiseEvaluatorDataset`
```

Comment:
So it is meant to evaluate pairs of evaluator responses? Is this any different than evaluating pairs of responses from any task?

Reply:
Oh so here we're evaluating the evaluator, which is tasked to evaluate 2 LLM responses to a given query. The query and responses can come from any task.

The output of our evaluator still needs to be a valid `EvaluationResult`.

```
pairwise_evaluator = PairwiseComparisonEvaluator(...) 
eval_result: EvaluationResult = pairwise_evaluator.aevaluate(
     query=query,
     answer=answer,
     second_answer=second_answer
)
```

It is the `eval_result` that we essentially will evaluate (i.e., evaluating this evaluator lol) by comparing it to the corresponding `reference_eval_result`